designer/client/src/components/graph/node-modal/fragment-input-definition/settings/variants/fields/UserDefinedListInput.tsx

```diff
@@ -3,22 +3,36 @@ import { EditableEditor } from "../../../../editors/EditableEditor";
 import { ExpressionLang } from "../../../../editors/expression/types";
 import AceEditor from "react-ace";
 import { ListItems } from "./ListItems";
-import React, { useState } from "react";
+import React, { useMemo, useState } from "react";
 import { useTranslation } from "react-i18next";
-import { FixedValuesOption, onChangeType } from "../../../item";
-import { VariableTypes } from "../../../../../../../types";
+import { FieldName, FixedValuesOption, onChangeType } from "../../../item";
+import { ReturnedType, VariableTypes } from "../../../../../../../types";
+import { Error, uniqueValueValidator, validators } from "../../../../editors/Validators";
+import HttpService from "../../../../../../../http/HttpService";
+import { useSelector } from "react-redux";
+import { getProcessProperties } from "../../../../NodeDetailsContent/selectors";
+import { getProcessId, getProcessToDisplay } from "../../../../../../../reducers/selectors/graph";
+import { GenericValidationRequest } from "../../../../../../../actions/nk/genericAction";
+import { debounce } from "lodash";
+import { EditorType } from "../../../../editors/expression/Editor";
 
 interface Props {
     onChange: (path: string, value: onChangeType) => void;
     path: string;
     fixedValuesList: FixedValuesOption[];
     variableTypes: VariableTypes;
     readOnly: boolean;
+    errors: Error[];
+    typ: ReturnedType;
+    name: string;
 }
 
-export const UserDefinedListInput = ({ fixedValuesList, path, onChange, variableTypes, readOnly }: Props) => {
+export const UserDefinedListInput = ({ fixedValuesList, path, onChange, variableTypes, readOnly, errors, typ, name }: Props) => {
     const [temporaryListItem, setTemporaryListItem] = useState("");
```

Comment:
After changing node name or when doing drag and drop, the value in `add list item` resets to empty string.

Reply:
Seems like there are two separate cases:
1. After changing the node name -> Validation error is clear but the value stays as it is,  it's a problem with the field name when we change the name of the param, the field name changes, and errors are reset by EditableEditor editor. Seems like, using uuid in the fieldName, should fix the issue. So we should leave it for now
2. doing drag and drop -> It's an error related to the drag and drop library. After DnD the new react component is created and the whole state is recreated. There is no good fix for it https://github.com/atlassian/react-beautiful-dnd/issues/850. The alternative to the local state is to use the global state, which is a little strange in this context ;). I would say, since it's an edge case, let's leave it for now.

WDYT?

Reply:
Thanks for clarification, let's leave this for another change